public/video-ui/src/util/hasUnpublishedChanges.js

```diff
@@ -0,0 +1,35 @@
+export function hasUnpublishedChanges(previewVideo, publishedVideo) {
+  if (!previewVideo) {
+    return false;
+  }
+
+  if (!publishedVideo) {
+    return true;
+  }
+
+  const previewPoster = previewVideo.posterImage;
+  const publishedPoster = publishedVideo.posterImage;
+  const previewAssets = previewVideo.assets;
+  const publishedAssets = publishedVideo.assets;
+
+  if (
```

Comment:
If we're doing stuff like this, might it be worth just doing a JSON.stringify and comparing the results? I'm slightly worried that we might add a field in the future and forget to change this code